Automattic/Gravatar-SDK-iOS
# Merge Ratings into AtlText

### Description

Now that #589 (QE: Avatar Rating) PR just merged into `trunk`, this merges `trunk` back into #590 (AltText Editor).

- merges `trunk` (including the new Avatar Rating capability)
- Adds `.altText` into the new Menu structure
- Renames `setRating(_ rating: AvatarRating, for avatar: AvatarImageModel) async` --> `update(_ avatar: AvatarImageModel, rating: AvatarRating) async -> Bool`
- Makes the `update()` functions `@discardableResult`
   - `AvatarService.update(::::)`
   - `AvatarPickerViewModel.update(_: altText)`
   - `AvatarPickerViewModel.update(_: rating)`
- Removes the `ProfileService.setRating(:::)` function in favor of the `AvatarService.update(::::)` function in the AltText PR
- Updates related unit tests

### Testing Steps

- Make sure the AltText changes are still intact
- Make sure the AltText and Rating menu items work as expected
